### PR TITLE
feat: enable groups env variable

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/8703

### Description:

- adds env variable `GROUPS_ENABLED`
- removes feature flag `group-management`
- documents env variable in docs

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -38,6 +38,8 @@ DBT_PROJECT_DIR=.
 
 SCHEDULER_ENABLED=true
 
+GROUPS_ENABLED=true
```
Comment (IrakliJani) [inline .env.development:41]: enabling locally

File: docker-compose.yml [hunk 2]
```diff
@@ -38,6 +38,7 @@ services:
       - HEADLESS_BROWSER_PORT=3000
       - RUDDERSTACK_WRITE_KEY=${RUDDERSTACK_WRITE_KEY}
       - SCHEDULER_ENABLED=true
+      - GROUPS_ENABLED=${GROUPS_ENABLED:-false}
```
Comment (IrakliJani) [inline docker-compose.yml:41]: disabling for images

File: docker/docker-compose.dev.yml [hunk 3]
```diff
@@ -32,6 +32,7 @@ services:
       - HEADLESS_BROWSER_HOST=headless-browser
       - HEADLESS_BROWSER_PORT=3000
       - SCHEDULER_ENABLED=true
+      - GROUPS_ENABLED=true
```
Comment (IrakliJani) [inline docker/docker-compose.dev.yml:35]: enabling for local docker compose

File: packages/backend/src/services/HealthService/HealthService.mock.ts [hunk 4]
```diff
@@ -62,4 +63,7 @@ export const Config = {
             loginPath: '',
         },
     },
-} as LightdashConfig;
+    groups: {
+        enabled: false,
+    },
+} as LightdashConfig; // TODO: Refactor this to be a mock of the actual configuration, rather than a partial that might contain incorrect properties.
```
Comment (IrakliJani) [inline packages/backend/src/services/HealthService/HealthService.mock.ts:69]: this is bad. it's used as a mock. messes up code in HealthService for tests

File: packages/backend/src/services/UserService.ts [hunk 5]
```diff
@@ -397,6 +397,7 @@ export class UserService {
             });
 
             if (
+                lightdashConfig.groups.enabled === true &&
```
Comment (IrakliJani) [inline packages/backend/src/services/UserService.ts:400]: if groups are not enabled we should not group sync even tho enableGroupSync might be present

File: render.yaml [hunk 6]
```diff
@@ -54,3 +54,5 @@ services:
           type: pserv
           name: headless-browser
           property: host
+      - key: GROUPS_ENABLED
+        value: true
```
Comment (IrakliJani) [inline render.yaml:58]: Is this the correct place to enable groups for preview environments?

File: docs/docs/references/groups.mdx [hunk 7]
```diff
@@ -8,6 +8,12 @@ To view and manage your groups, go to the `Organization Settings` > `Users & gro
 
 <img src={GroupsManagementPage} width="1230" height="520" style={{display: "block", margin: "0 auto 20px auto"}}/>.
 
+:::info
```
Comment (owlas) [inline docs/docs/references/groups.mdx:11]: in another PR let's move this to pills at the top of the page

**Review by owlas — APPROVED** [review]
Great work. But we need another PR to conditionally mount the groups api routes

File: docs/docs/references/groups.mdx (hunk 7 above)
Comment (IrakliJani) [reply docs/docs/references/groups.mdx:7]: done in this PR to resolve conflict with this PR https://github.com/lightdash/lightdash/pull/8727#issuecomment-1910263460

doc preview: https://deploy-preview-8704--peaceful-bassi-cbf284.netlify.app/references/groups

**Pull request merged**